server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamLifecycle.java

```diff
@@ -40,11 +40,15 @@
 /**
  * Holds the data stream lifecycle metadata that are configuring how a data stream is managed. Currently, it supports the following
  * configurations:
+ * - enabled
  * - data retention
  * - downsampling
  */
 public class DataStreamLifecycle implements SimpleDiffable<DataStreamLifecycle>, ToXContentObject {
 
+    // Versions over the wire
+    public static final TransportVersion ADDED_ENABLED_FLAG_VERSION = TransportVersion.V_8_500_051;
```

Comment:
This is nice ! :)

Reply:
Glad you like it, it definitely helps now when updating the versions, and I think it will help when we clean up or merge versions in the future as they go unused.